File: TD/TD4/Code/P1.cpp
/* Calculer et retourner si existent les racines d'un polynôme du 2 degré */
#include <cmath>
#include <iostream>

using namespace std;

/* Procédure */

void racine(float a, float b, float c, float &x1, float &x2)
{
  float d;
  
  d = b * b - 4 * a * c;
  
  if (d > 0) {
    x1 = (-b - sqrt(d)) / (2 * a);
    x2 = (-b + sqrt(d)) / (2 * a);
  } else if (d == 0) {
    x1 = x2 = -b / (2 * a);
  } else {
    x1 = x2 = NAN;
  }
}

/* Utilisation */

int main()
{
  float x1;
  float x2;
  
  racine(3, 5, 7, x1, x2);
  
  cout << "x1 = " << x1 << "\n" << "x2 = " << x2 << endl;
  return 0;
}

File: TD/TD4/Code/Python/P1.py
#!/usr/bin/python
# Calculer et retourner si existent les racines d'un polynôme du 2 degré
from math import sqrt, nan


def racine(a, b, c):
    d = b ** 2 - 4 * a * c

    if d > 0:
        x1 = (-b - sqrt(d)) / (2 * a)
        x2 = (-b + sqrt(d)) / (2 * a)
    elif d == 0:
        x1 = x2 = -b / (2 * a)
    else:
        x1 = x2 = nan

    return x1, x2


print('Donne moi trois valeurs : ', end='')
a = int(input())
b = int(input())
c = int(input())

x1, x2 = racine(a, b, c)

print('x1 = ', x1, '\nx2 = ', x2, sep='')
